packages/mask/background/services/backup/restore.ts

```diff
@@ -19,12 +19,18 @@ export interface RestoreUnconfirmedBackupOptions {
     action: 'confirm' | 'wallet'
 }
 
-export async function restoreUnconfirmedBackup({ id, action }: RestoreUnconfirmedBackupOptions): Promise<void> {
+export async function restoreUnconfirmedBackup(
+    { id, action }: RestoreUnconfirmedBackupOptions,
+    // should pass true when restore for merge backup
+    temporary?: boolean,
+): Promise<void> {
     const backup = unconfirmedBackup.get(id)
     if (!backup) throw new Error('Backup not found')
 
-    const granted = await requestHostPermission(backup.settings.grantedHostPermissions)
-    if (!granted) return
+    if (!temporary) {
```

Comment:
I was wondering if the permission request might be required. No prompt dialog will appear if the user has approved all of the host permissions.

Reply:
if we merge it without requiring necessary permissions, it might not work properly.